src/core/unit.c

```diff
@@ -3210,7 +3250,7 @@ int unit_install_bus_match(Unit *u, sd_bus *bus, const char *name) {
                          "member='NameOwnerChanged',"
                          "arg0='", name, "'");
 
-        return sd_bus_add_match_async(bus, &u->match_bus_slot, match, signal_name_owner_changed, NULL, u);
+        return sd_bus_add_match_async(bus, &u->match_bus_slot, match, signal_name_owner_changed, signal_name_owner_changed_installed, u);
```

Comment:
nah, don't wait for the match to finish installing. just enqueue the GetNameOwner method call asynchonrously right after enqueing the add match call